tests/test_modeling_common.py

```diff
@@ -652,27 +639,9 @@ def test_training_gradient_checkpointing_use_reentrant_false(self):
                 or not model_class.supports_gradient_checkpointing
             ):
                 continue
-
-            model = model_class(config)
-            model.to(torch_device)
-            model.gradient_checkpointing_enable(gradient_checkpointing_kwargs={"use_reentrant": False})
-            model.train()
-
-            # unfreeze additional layers
-            for p in model.parameters():
-                p.requires_grad_(True)
-
-            optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
-
-            inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
-            loss = model(**inputs).loss
-            loss.backward()
-
-            optimizer.step()
-
-            for k, v in model.named_parameters():
-                if v.requires_grad:
-                    self.assertTrue(v.grad is not None, f"{k} in {model_class.__name__} has no gradient!")
+            self.check_training_gradient_checkpointing(
+                model_class, gradient_checkpointing_kwargs={"use_reentrant": False}
+            )
```

Comment:
maybe put what above to `check_training_gradient_checkpointing`, and no need to have/pass  the argument `model_class`. This way, save even more code lines.

Reply:
Done!